### PR TITLE
Fix navigation error in tool shed tests.

### PR DESCRIPTION
**Pull request by jmchilton**
I'm pretty sure I correctly identified the problem in comments and I've fixed it two different ways.

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by jmchilton** [comment]
Got lost on a red herring at first - I think all the problems are just the navigation issues. I think what is happening here is the logout has a race condition. The failures seem to happen on the second login within two of the same test method. I think the logout redirect with conflicting with the goto. There are a number of ways to work around this - I'm not sure it is worth putting too much effort into getting it perfect - certainly the tests that initialize different sessions for different logins are going to be entirely robust against this.

**Comment by jmchilton** [comment]
My guess is https://github.com/galaxyproject/galaxy/blob/dev/lib/tool_shed/webapp/frontend/src/stores/auth.store.ts#L49 is the frontend redirect that is conflicting with the playwright commands.

**Review by mvdbeek — APPROVED** [review]
Looks great to me, just needs a `make format`.

File: lib/tool_shed/test/functional/test_0030_repository_dependency_revisions.py [hunk 1]
```diff
@@ -116,6 +116,9 @@ def test_0030_generate_repository_dependencies_for_emboss_5(self):
     def test_0035_generate_repository_dependencies_for_emboss_6(self):
         """Generate a repository_dependencies.xml file specifying emboss_datatypes and upload it to the emboss_6 repository."""
         emboss_6_repository = self._get_repository_by_name_and_owner(emboss_6_repository_name, common.test_user_1_name)
+        assert (
+            emboss_6_repository
+        ), f"Failed to find repository for {common.test_user_1_name}/{emboss_6_repository_name}"
```
Comment (nsoranzo) [inline lib/tool_shed/test/functional/test_0030_repository_dependency_revisions.py]: This won't be needed any more with the changes in #16901 , drop?

Comment (jmchilton) [reply lib/tool_shed/test/functional/test_0030_repository_dependency_revisions.py]: It wasn't helping anyway - the real issue with the login failure. Sure I'll drop this.

**Pull request merged**